paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -396,6 +434,86 @@ def sync_secrets(
     return True
 
 
+def sync_database_credentials(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Dict[str, str],
+    soa_dir: str,
+    vault_token_file: str,
+) -> bool:
+    config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
+    system_paasta_config = load_system_paasta_config()
+    database_credentials_vault_overrides = (
+        system_paasta_config.get_database_credentials_vault_overrides()
+    )
```

Comment:
thinking about this more: can we set a default value for this? or should we add a check here that `database_credentials_vault_overrides` is not empty?

otherwise, what would happen if this gets called without the overrides?

Reply:
might be worth testing the empty dict case in dev to make sure it won't cause chaos should this get deployed to prod without this config :)

(or crash early here if this is an empty dict)

Reply:
I tested it - empty dict doesn't crash but the script crashes anyway because it cannot connect to Vault. ie the provider calls are no-ops (`self.clients[self.ecosystems[0]].read` <-- becomes `None.read`, throws exception)

this is a good error though? given it's in a CRON, not like it'll crash PaaSTA services. or should it fail gracefully on no-connection to Vault?

Reply:
given that this script is doing somewhat non-standard things (i.e., not reading from our normal vault), i'd probably err towards the error coming from somewhere closer to the misconfiguration - otherwise, someone getting paged would need to follow the code all the way up to this overrides section and realize that something has gone wrong with the config

Reply:
technically the error started at the `provider = get_secret_provider(` line since it doesn't do a test connection imo. maybe that's where it should go ?

otherwise, what I can do is something like:

```
 client = self.clients[self.ecosystems[0]]

 if client is None:
    return None <-- this happens during invalid connections since self.clients['devc'] is set to None
 result = client.read(...)
```

I'll lean towards the second approach but maybe long term, the providers have a "test connection" check?

Reply:
++ added a check for client is not None. will leave this unresolved to give you a chance to assess (though these "outdated" code blocks are quite confusing in github)

Reply:
raised an exception on client is None in another thread. resolving this one